cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandlerTest.java

```diff
@@ -144,7 +144,6 @@ public void testBasics() throws Exception {
       // verify creation of dataset instance with null properties
       Assert.assertEquals(HttpStatus.SC_OK, createInstance("nullPropertiesTable", "datasetType2").getResponseCode());
 
-    } finally {
```

Comment:
This unit-test was misbehaved: If any failure occurred early in the test, then the asserts on deleteInstance() calls in this finally block would fail, and that failure becomes the only exception reported for this unit test, masking the original exception. Moving the asserts before the finally block and doing the cleanup without asserting their status in the finally clause fixes that.